src/html.c

```diff
@@ -59,17 +59,31 @@ static void filter_html_block(cmark_html_renderer *renderer, uint8_t *data, size
     cmark_strbuf_put(html, data, (bufsize_t)len);
 }
 
-static bool S_put_footnote_backref(cmark_html_renderer *renderer, cmark_strbuf *html) {
+static bool S_put_footnote_backref(cmark_html_renderer *renderer, cmark_strbuf *html, cmark_node *node) {
   if (renderer->written_footnote_ix >= renderer->footnote_ix)
     return false;
   renderer->written_footnote_ix = renderer->footnote_ix;
 
-  cmark_strbuf_puts(html, "<a href=\"#fnref");
-  char n[32];
-  snprintf(n, sizeof(n), "%d", renderer->footnote_ix);
-  cmark_strbuf_puts(html, n);
+  cmark_strbuf_puts(html, "<a href=\"#fnref:");
+  cmark_strbuf_put(html, node->as.literal.data, node->as.literal.len);
```

Comment:
Isn't this string user controlled?  Maybe I'm missing something but I would expect sanitization to prevent both:
 
- chars that are not legal as an anchor
- malicious input that attempts to escape the attribute quote or tag

Maybe the set of chars allowed in labels is already very constrained, can you point me to where that's defined?

Reply:
You know, I thought about this at the time and largely shrugged because we perform sanitization after the fact, and I'd _hoped_ that the footnote label was being sanitized, and so I hadn't actually checked.

```
echo -e 'hello [^"><script>alert(1)</script>]\n\n[^"><script>alert(1)</script>]: foo' |./build/src/cmark-gfm -e footnotes -e autolink -e strikethrough -e table

<p>hello <sup class="footnote-ref"><a href="#fn:"><script>alert(1)</script>" id="fnref:"><script>alert(1)</script>">1</a></sup></p>
<section class="footnotes">
<ol>
<li id="fn:"><script>alert(1)</script>">
<p>foo <a href="#fnref:"><script>alert(1)</script>" class="footnote-backref">↩</a></p>
</li>
</ol>
</section>
```

So in terms of impact we'd be mitigated but it's still _unsatisfying_.

Considering we _do_ perform adequate sanitization elsewhere, this shouldn't be too difficult? (famous last words) to change.

```
echo '!["alt](/foo "title")' |./build/src/cmark-gfm -e footnotes -e autolink -e strikethrough -e table

<p><img src="/foo" alt="&quot;alt" title="title" /></p>
```

Thanks for calling this out.

Reply:
Turns out we had a really handy `href` escape library, so this was an easy swap.

I don't understand why you'd ever want to have unsupported characters in the href??? so I said screw it, and always perform the sanitization even in `--unsafe` mode.